libsigopt/aux/adapter_info_containers.py

```diff
@@ -9,21 +9,23 @@
 @dataclass(slots=True)
 class PointsContainer:
   points: numpy.ndarray
-  values: numpy.ndarray = None
-  value_vars: numpy.ndarray = None
-  failures: numpy.ndarray = None
-  task_costs: numpy.ndarray = None
+  values: numpy.ndarray | None = None
+  value_vars: numpy.ndarray | None = None
+  failures: numpy.ndarray | None = None
+  task_costs: numpy.ndarray | None = None
 
 
 @dataclass(slots=True)
 class DomainInfo:
   constraint_list: list
-  domain_components: list = None
+  domain_components: list | None = None
   force_hitandrun_sampling: bool = False
-  priors: list = None
+  priors: list | None = None
 
   @property
   def dim(self) -> int:
+    if self.domain_components is None:
+      return 0
```

Comment:
can we raise an `ValueError` here? I can also do it a different PR. Let me know what you prefer